File: Day7/StockBuySell.py
def StockBuySell(prices):
    minsofar=prices[0]
    res=0
    n=len(prices)
    for i in range(1,n):
        minsofar=min(minsofar,prices[i])
        res=max(res,prices[i]-minsofar)
    return res;
        
        
prices = [7,10,1,3,6,9,2]
print(StockBuySell(prices))

File: Day7/StockBuySell.cpp
#include<bits/stdc++.h>
using namespace std;

int StockBuySell(vector<int> prices, int n){

    int res=0;
    int minsofar=prices[0];
    for(int i=1;i<n;i++){
        minsofar = min(minsofar,prices[i]);
        res=max(res,prices[i]-minsofar);
    }

    return res;


}

int main(){
    vector<int> prices = {7,10,1,3,6,9,2};
    int n= prices.size();

    int r = StockBuySell(prices,n);
    cout<<"result:"<<r<<endl;
};